portfolio/src/main/java/com/google/sps/servlets/MapMarkers.java

```diff
@@ -0,0 +1,80 @@
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.gson.sps.data.MapMarker;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that adds MapMarkers to a database */
+@WebServlet("/markers")
+public class MapMarkers extends HttpServlet {
+    private static final String MARKER_TABLE_NAME = "MARKER_LOCATION";
+    private static final String MARKER_LAT_COLUMN_NAME = "LATITUDE";
+    private static final String MARKER_LNG_COLUMN_NAME = "LONGITUDE";
+    private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        List<Marker> markersToDisplay = new ArrayList<>();
+        Query query = new Query(MARKER_TABLE_NAME);
+        PreparedQuery results = datastore.prepare(query);
+        for (Entity entity : results.asIterable()) {
+            MapMarker marker = new MapMarker((int) entity.getProperty(MARKER_LAT_COLUMN_NAME),
+                                             (int) entity.getProperty(MARKER_LNG_COLUMN_NAME)); 
+            markersToDisplay.add(marker);
+        }
+        String markerJSON = convertToJson(markersToDisplay);
+        
+        // Send JSON response 
+        response.setContentType("application/json;");
+        response.getWriter().println(markerJSON);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+         String lat_str = request.getParameter("lat");
+         String lng_str = request.getParameter("lng");
+         if ((lat_str != null && !lat_str.isEmpty()) && 
+              (lng_str != null && !lng_str.isEmpty())) 
+            {
+                int lat = Integer.parseInt(lat_str);
+                int lng = Integer.parseInt(lng_str);
+                Entity coordinates = new Entity(MARKER_TABLE_NAME);
+                coordinates.setProperty(MARKER_LAT_COLUMN_NAME, lat);
+                coordinates.setProperty(MARKER_LNG_COLUMN_NAME, lng);
+                datastore.put(coordinates);
+            }
+         response.sendRedirect("/index.html");
+    }
+
+    private static String convertToJson(List<MapMarker> markersToDisplay) {
+        Gson gson = new Gson();
+        String json = gson.toJson(comments);
```

Comment:
Did you mean `markersToDisplay` instead of `comments`?

Reply:
yes! lol I have no idea why this didn't push but I also fixed this locally-will push soon! thank you!